@navikt/core/react/src/form/combobox/FilteredOptions/filteredOptionsContext.tsx

```diff
@@ -0,0 +1,278 @@
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  createContext,
+  useContext,
+  useCallback,
+  useRef,
+} from "react";
+import { useCustomOptionsContext } from "../customOptionsContext";
+import { useInputContext } from "../Input/inputContext";
+import usePrevious from "../../../util/usePrevious";
+
+const normalizeText = (text: string): string =>
+  typeof text === "string" ? `${text}`.toLowerCase().trim() : "";
+
+const isPartOfText = (value, text) =>
+  normalizeText(text).startsWith(normalizeText(value ?? ""));
+
+const isValueInList = (value, list) =>
+  list?.find((listItem) => normalizeText(value) === normalizeText(listItem));
+
+const getMatchingValuesFromList = (value, list) =>
+  list?.filter((listItem) => isPartOfText(value, listItem));
+
+type FilteredOptionsContextType = {
+  activeDecendantId?: string;
+  ariaDescribedBy?: string;
+  filteredOptionsRef: React.RefObject<HTMLUListElement>;
+  filteredOptionsIndex: number | null;
+  setFilteredOptionsIndex: (index: number) => void;
+  isListOpen: boolean;
+  setInternalListOpen: (open: boolean) => void;
+  isLoading?: boolean;
+  filteredOptions: string[];
+  isValueNew: boolean;
+  toggleIsListOpen: (newState?: boolean) => void;
+  currentOption: string | null;
+  resetFilteredOptionsIndex: () => void;
+  moveFocusUp: () => void;
+  moveFocusDown: () => void;
+  moveFocusToInput: () => void;
+  moveFocusToEnd: () => void;
+};
+const FilteredOptionsContext = createContext<FilteredOptionsContextType>(
+  {} as FilteredOptionsContextType
+);
+
+export const FilteredOptionsProvider = ({ children, value: props }) => {
+  const {
+    allowNewValues,
+    isListOpen: isExternalListOpen,
+    isLoading,
+    options,
+    singleSelect,
+  } = props;
+  const filteredOptionsRef = useRef<HTMLUListElement | null>(null);
+  const {
+    inputProps: { id },
+    value,
+    searchTerm,
+    setValue,
+    setSearchTerm,
+    shouldAutocomplete,
+  } = useInputContext();
+  const [ariaDescribedBy, setAriaDescribedBy] = useState<string>();
+
+  const [filteredOptionsIndex, setFilteredOptionsIndex] = useState<
+    number | null
+  >(null);
+  const [isInternalListOpen, setInternalListOpen] = useState(false);
+  const { customOptions } = useCustomOptionsContext();
+
+  const filteredOptions = useMemo(() => {
+    const opts = [...customOptions, ...options];
+    setFilteredOptionsIndex(null);
+    return getMatchingValuesFromList(searchTerm, opts);
+  }, [customOptions, options, searchTerm]);
+
+  useEffect(() => {
+    if (isExternalListOpen !== undefined)
+      setInternalListOpen(isExternalListOpen);
+  }, [isExternalListOpen]);
```

Comment:
```suggestion
  useEffect(() => isExternalListOpen !== undefined && setInternalListOpen(isExternalListOpen)

  , [isExternalListOpen]);
```

Reply:
Replaced this useEffect with a useMemo, that evaluates to `isExternalListOpen ?? isInternalListOpen`.